lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -674,6 +730,83 @@ public void OnNext(IFailedTask failedTask)
         }
         #endregion IFailedTask
 
+        private void TimeoutMonitor(object source, ElapsedEventArgs e)
+        {
+            Logger.Log(Level.Info, "Entering TimeoutMonitor at {0}", DateTime.Now);
+            lock (_lock)
+            {
+                switch (_systemState.CurrentState)
+                {
+                    // TODO: Handle time out if ActiveContexts are not received in timeout limit
+                    case SystemState.WaitingForEvaluator:
+                        break;
+
+                    // TODO: Handle time out if RunningTasks are not received in timeout limit
+                    case SystemState.SubmittingTasks:
+                        break;
+
+                    // TODO: Handle time out if CompletedTasks are not received in timeout limit
+                    case SystemState.TasksRunning:
+                        break;
+
+                    // Handle timeout for closing tasks
+                    case SystemState.ShuttingDown:
+                        Logger.Log(Level.Info, "_taskManager.AverageClosingTime {0}, _minTaskWaitingForCloseTimeout: {1}", _taskManager.AverageClosingTime(), _minTaskWaitingForCloseTimeout);
+                        int taskClosingTimeout = Math.Max(_minTaskWaitingForCloseTimeout, _taskManager.AverageClosingTime() * TaskWaitingForCloseTimeFactor);
+                        var waitingTasks = _taskManager.TasksWaitingForClose(taskClosingTimeout);
+
+                        if (waitingTasks.Count > 0)
```

Comment:
`waitingTasks.Any()` will be a bit better, since we don't need the actual count, just the fact that there are some tasks